# Q1: 02-Aug-2022

Please refer to the Universities.xlsx on GitHub (https://github.com/Accelerate-AI/UnsupervisedLearning_Clustering/tree/main/DailyQuestions/Universities.xlsx). 

Find out the Euclidean distance between Stanford and MIT.


In [1]:
import pandas as pd
import numpy as np

In [2]:
col_list = ['Univ', 'SAT', 'Top10', 'Accept', 'SFRatio', 'Expenses', 'GradRate']
df = pd.read_excel("Universities.xlsx", usecols=col_list)
df.head(2)

Univ   SAT  Top10  Accept  SFRatio  Expenses  GradRate
0    Brown  1310     89      22       13     22704        94
1  CalTech  1415    100      25        6     63575        81

In [3]:
series_Stanford = df[df['Univ'] == "Stanford"].copy().drop('Univ', axis = 1).iloc[0]
series_MIT = df[df['Univ'] == "MIT"].copy().drop('Univ', axis = 1).iloc[0]

In [4]:
def get_euclidean_distince(series_1, series_2):
    return np.round(np.sqrt(np.sum((series_1 - series_2) ** 2)), 2)

In [5]:
euc_dist = get_euclidean_distince(series_Stanford, series_MIT)
print(f"Euclidean distance is : {euc_dist}")

Euclidean distance is : 1580.17


In [6]:
# series_Stanford
# series_MIT
# (series_Stanford - series_MIT) ** 2

# Answer:

    1. The Euclidean distance between Stanford and MIT is  1580.17
    2. We also observe that entire distance is based on Expenses value. To overcome this, we should try to normalize/standardize features, before calculaitng Euclidean distance